clojure-mode.el

```diff
@@ -1534,6 +1546,197 @@ This will skip over sexps that don't represent objects, so that ^hints and
           (backward-sexp 1))
         (setq n (1- n))))))
 
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;
+;; Refactoring support
+;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+(defcustom clojure-thread-all-but-last nil
+  "When true `cljr-thread-first-all' and `cljr-thread-last-all' don't thread the last expression."
+  :package-version '(clojure-mode . "5.3.0")
+  :safe #'booleanp
+  :type 'boolean)
+
+(defun clojure--unwind-last ()
+  (forward-sexp)
+  (save-excursion
+    (let ((contents (clojure-delete-and-extract-sexp)))
+      (when (looking-at " *\n")
+        (join-line -1))
+      (clojure--ensure-parens-around-function-names)
+      (let* ((sexp-beg-line (line-number-at-pos))
+             (sexp-end-line (progn (forward-sexp)
+                                   (line-number-at-pos)))
+             (multiline-sexp-p (not (= sexp-beg-line sexp-end-line))))
+        (down-list -1)
+        (when multiline-sexp-p
+          (newline))
+        (insert contents)
+        (when multiline-sexp-p
+          (clojure-indent-line)))))
+  (forward-char))
+
+(defun clojure--wrap-around ()
+  (insert (format "(%s)" (clojure-delete-and-extract-sexp))))
+
+(defun clojure--ensure-parens-around-function-names ()
+  (unless (looking-at "[\n\r\t ]?(")
+    (skip-syntax-forward " ")
+    (clojure--wrap-around)
+    (backward-sexp)))
+
+(defun clojure--unwind-first ()
+  (forward-sexp)
+  (save-excursion
+    (let ((contents (clojure-delete-and-extract-sexp)))
+      (when (looking-at " *\n")
+        (join-line -1))
+      (clojure--ensure-parens-around-function-names)
+      (down-list)
+      (forward-sexp)
+      (insert contents)))
+  (forward-char))
+
+(defun clojure--pop-out-of-threading ()
+  (down-list 2)
+  (backward-up-list)
+  (raise-sexp)
+  (save-excursion
+    (let ((beg (point))
+          (end (progn
+                 (forward-sexp)
+                 (point))))
+      (clojure-indent-region beg end))))
+
+(defun clojure--nothing-more-to-unwind ()
+  (save-excursion
+    (let ((beg (point)))
+      (forward-sexp)
+      (down-list -1)
+      (backward-sexp 2) ;; the last sexp, the threading macro
+      (when (looking-back "(\\s-*")
+          (backward-up-list)) ;; and the paren
+      (= beg (point)))))
+
+;;;###autoload
+(defun clojure-unwind ()
+  "Unwind thread at point or above point by one level.
+Return nil if there are no more levels to unwind."
+  (interactive)
+  (ignore-errors
+    (forward-char 3))
```

Comment:
`forward-char 3` might not work as expected for longer macros like `some->>`.
I would rather do 
```
(when (looking-at "(")
  (forward-char 1)
  (forward-sexp 1))
```